### DISCUSSION TITLE
Manual training of models for multi-step direct forecasting

### DISCUSSION THREAD

**Discussion by laxhammar** [Nixtla/mlforecast#304] [Q&A] [answered]
Hi!

I'm experimenting with multi-step direct forecasting. `MLForecast.fit` supports this through setting `max_horizon`, which will train multiple models under the hood.  However I'm trying to figure out how I can do something similar in case of manual feature engineering/training, ie when using `MLForecast.preprocess`? There is `MLForecast.fit_models` but it does not seem to support multi-step direct forecasting models by `max_horizon` or similar? Is there some other function I can use? Could anyone point me to an example/tutorial where this is done?

**Comment by jmoralez** [comment, marked as answer]
Hey @laxhammar, thanks for using mlforecast. It's possible to use `MLForecast.fit_models`, that method checks if the target is a matrix and if it is it trains one model for each column. Here's a minimal example:

```python
from mlforecast import MLForecast
from mlforecast.utils import generate_series
from sklearn.linear_model import LinearRegression

series = generate_series(5)
fcst = MLForecast(
    models=[LinearRegression()],
    freq='D',
    lags=[1],
)
X, y = fcst.preprocess(series, max_horizon=4, return_X_y=True)
fcst.fit_models(X, y)
assert len(fcst.models_['LinearRegression']) == 4
```

**Reply by laxhammar** [reply]
Hi José, thanks a lot! This solved my problem!
